### PR TITLE
generate a single caching hash

### PR DESCRIPTION
**Pull request by rarmatei**
Fixes https://linear.app/nxdev/issue/INF-1095/agents-workflows-will-fail-if-more-than-one-package-lockyarnlock

### CODE REVIEW

File: workflow-steps/cache/hashing-utils.ts [hunk 1]
```diff
@@ -40,5 +36,14 @@ export function hashKey(key: string): string {
   const globHashes = globsToHash.map((globPattern) => {
     return hashFileContents(globPattern);
   });
-  return [...hardcodedKeys, ...globHashes].join(' | ');
+  const hashCollections = [...hardcodedKeys, ...globHashes];
+
+  if (hashCollections.length > 1) {
+    return hash(hashCollections.join(' | '));
+  }
+  return hashCollections.join(' | ');
```
Comment (rarmatei) [inline workflow-steps/cache/hashing-utils.ts:44]: I could've just always returned 
```typescript
return hash(hashCollections.join(' | '));
```

but I kept the old way of hashing when it's a single-key for backward compatibility purposes - so we don't bust everyone's cache when this gets merged in

File: workflow-steps/cache/output/main.js [hunk 2]
```diff
@@ -5978,7 +5978,14 @@ function hashKey(key2) {
   const globHashes = globsToHash.map((globPattern) => {
     return hashFileContents(globPattern);
   });
-  return [...hardcodedKeys, ...globHashes].join(" | ");
+  const hashCollections = [...hardcodedKeys, ...globHashes];
+  if (hashCollections.length > 1) {
+    return hash(hashCollections.join(" | "));
+  }
+  return hashCollections.join(" | ");
+}
```
Comment (lourw) [inline workflow-steps/cache/output/main.js:5986]: nit: formatting lol, and on the post.js file too
```suggestion
}

```

File: workflow-steps/cache/output/post.js [hunk 3]
```diff
@@ -5966,7 +5966,14 @@ function hashKey(key) {
   const globHashes = globsToHash.map((globPattern) => {
     return hashFileContents(globPattern);
   });
-  return [...hardcodedKeys, ...globHashes].join(" | ");
+  const hashCollections = [...hardcodedKeys, ...globHashes];
+  if (hashCollections.length > 1) {
+    return hash(hashCollections.join(" | "));
+  }
+  return hashCollections.join(" | ");
+}
```
Comment (lourw) [inline workflow-steps/cache/output/post.js:5974]:
```suggestion
}

```

Comment (rarmatei) [reply workflow-steps/cache/output/post.js:5974]: these are auto-generated from the TS build 😢

File: workflow-steps/cache/hashing-utils.ts (hunk 1 above)
Comment (lourw) [reply workflow-steps/cache/hashing-utils.ts:44]: I'd leave that in a comment in the code?

File: workflow-steps/cache/output/main.js (hunk 2 above)
Comment (rarmatei) [reply workflow-steps/cache/output/main.js:5986]: it's auto-generated from the build command 😢

Comment (lourw) [reply workflow-steps/cache/output/main.js:5986]: Oh woops, did not realize. Thought I was still looking at the .ts files.

**Review by lourw — APPROVED** [review]

**Pull request merged**